apollo-compiler/src/main/kotlin/com/apollostack/compiler/ir/GraphQlType.kt

```diff
@@ -0,0 +1,49 @@
+package com.apollostack.compiler.ir
+
+import com.squareup.javapoet.ClassName
+import com.squareup.javapoet.ParameterizedTypeName
+import com.squareup.javapoet.TypeName
+
+sealed class GraphQlType(val nullable: Boolean) {
+
+  class GraphQlString(nullable: Boolean = true) : GraphQlType(nullable)
+
+  class GraphQlId(nullable: Boolean = true) : GraphQlType(nullable)
+
+  class GraphQlInt(nullable: Boolean = true) : GraphQlType(nullable)
+
+  class GraphQLFloat(nullable: Boolean = true) : GraphQlType(nullable)
+
+  class GraphQLBoolean(nullable: Boolean = true) : GraphQlType(nullable)
+
+  class GraphQLList(nullable: Boolean = true, val listType: GraphQlType) : GraphQlType(nullable)
+
+  class GraphQlUnknown(nullable: Boolean = true, val typeName: String) : GraphQlType(nullable)
+
+  fun toJavaTypeName(): TypeName {
+    return when (this) {
+      is GraphQlString -> ClassName.get(String::class.java)
+      is GraphQlId -> if (nullable) ClassName.get(java.lang.Long::class.java) else ClassName.LONG
+      is GraphQlInt -> if (nullable) ClassName.get(java.lang.Integer::class.java) else TypeName.INT
+      is GraphQLFloat -> if (nullable) ClassName.get(java.lang.Float::class.java) else TypeName.FLOAT
+      is GraphQLBoolean -> if (nullable) ClassName.get(java.lang.Boolean::class.java) else TypeName.BOOLEAN
```

Comment:
nit: do you need to use the fully qualified class names here (`java.lang.Klass`)? It's cleaner if you could omit that